### PR TITLE
upgrade babel 🍭🚀

### PR DESCRIPTION
**Pull request by adnasa**
oooh yeah!!!

#### Summary 

- upgrades babel 😎
- uses class prop initializers today, with static binding on methods.
- locally: build + build:prod passed, `dext` works as expected.

#### Tech debt

Unfortunately I couldn't get rid of `babel-core` in favor of `@babel/core` all thanks to the mighty jest requiring this, so the next dev looking into the `package.json` will naturally raise questions.

__Important note__

I've now added a requirement on the node-engine for the next dev to the have the minimum of `8.0.0`. I hope to push this even further soon, but for now this is good enough.

#### QA

Important steps as you review

1. remove `node_modules`
2. switch node version to `v8.x.x`
3. run all the yarns (install, build, start, dev, test)

the reason for this, is that sometimes `electron` imposes a checkSum error and just refuses to continue and blocks the entire installation process

### CODE REVIEW

File: .babelrc [hunk 1]
```diff
@@ -1,3 +1,13 @@
 {
-  "presets": ["react", "es2015", "stage-0"]
+  "presets": [
+    "@babel/preset-react",
+    "@babel/preset-stage-3",
+    "@babel/preset-es2017"
+  ],
+  "plugins": ["transform-class-properties"],
```
Comment (adnasa) [inline .babelrc]: I found this odd.

1. `transform-class-properties` is already part of `preset-stage-3`
2. tests breaks when this is omitted

Any ideas on how we pass tests without having to explicitly define this plugin in the `.babelrc`?

File: .babelrc [hunk 2]
```diff
@@ -1,3 +1,13 @@
...
+  "env": {
+    "test": {
+      "presets": ["@babel/preset-env", "@babel/preset-react"]
```
Comment (adnasa) [inline .babelrc]: OMG... of course... it's missing here 😂 
will try again

File: .babelrc (hunk 1 above)
Comment (vutran) [reply .babelrc]: That's actually in stage-2 (I think?)

**Comment by adnasa** [comment]
I can't find any further information on how this problem can be resolved.

```
[4/4] 📃  Building fresh packages...
[1/3] ⠁ electron
[-/3] ⠁ waiting...
[3/3] ⠁ weak: gyp info spawn args   '-Goutput_dir=.' ]
[-/3] ⠁ waiting...
error An unexpected error occurred: "/Users/adnasa/workspace/repos/dext/node_modules/electron: Command failed.
Exit code: 1
Command: sh
Arguments: -c node install.js
Directory: /Users/adnasa/workspace/repos/dext/node_modules/electron
Output:
/Users/adnasa/workspace/repos/dext/node_modules/electron/install.js:47
  throw err
  ^

Error: Generated checksum for \"electron-v1.8.4-darwin-x64.zip\" did not match expected checksum.
    at ChecksumMismatchError.ErrorWithFilename (/Users/adnasa/workspace/repos/dext/node_modules/sumchecker/build.js:41:124)
    at new ChecksumMismatchError (/Users/adnasa/workspace/repos/dext/node_modules/sumchecker/build.js:56:133)
    at Hash.<anonymous> (/Users/adnasa/workspace/repos/dext/node_modules/sumchecker/build.js:203:22)
    at emitNone (events.js:105:13)
    at Hash.emit (events.js:207:7)
    at emitReadable_ (_stream_readable.js:500:10)
    at emitReadable (_stream_readable.js:494:7)
    at addChunk (_stream_readable.js:261:7)
    at readableAddChunk (_stream_readable.js:237:11)
```

__edit__: so far, what I found out is that peeps got it working by downgrading... dafaq

**Comment by adnasa** [comment]
> edit: so far, what I found out is that peeps got it working by downgrading... dafaq

to node 6, I might add 😂

**Comment by vutran** [comment]
> to node 6, I might add 😂

I've been running `npm` only. I think we should ditch `yarn` since it's very cumbersome to keep both lock files in sync.

**Comment by adnasa** [comment]
I've been reading electron issues, so I never assumed that the problem is with yarn itself.

**Comment by adnasa** [comment]
I'm encountering this problem with `node 8.0.0` and above,
which is really odd and confusing, considering the `electron` itself promotes `8.2.1` on their website

**Comment by adnasa** [comment]
aight here we go, @vutran 462d13c

Context to others, we've discovered that it's problematic to run `yarn install` with the following environment, both locally and on travis:

```
- yarn >= 1.5.4
- node >= 8.0.0
- electron >= 1.8.4
```

Considering that we wouldn't want to downgrade node-version in favour of having "yarn working", we decided for now to ditch yarn and move on with `npm`

**Comment by adnasa** [comment]
passed build 😎 🔥 💻

File: package.json [hunk 3]
```diff
@@ -91,5 +93,8 @@
     "redux-saga": "^0.14.3",
     "screen-saver": "^0.2.1",
     "string_score": "^0.1.22"
+  },
+  "engines": {
+    "node": ">= 8.2.1"
```
Comment (adnasa) [inline package.json:98]: 🔥

File: package.json [hunk 4]
```diff
@@ -59,7 +61,7 @@
     "eslint-plugin-jsx-a11y": "6.0.3",
     "eslint-plugin-prettier": "2.6.0",
     "eslint-plugin-react": "7.7.0",
-    "jest": "^22.4.3",
+    "jest": "^21.3.0-beta.15",
```
Comment (adnasa) [inline package.json:64]: this beta version is required for babel-7 compatibility

File: package.json [hunk 5]
```diff
@@ -34,20 +34,22 @@
   },
   "license": "MIT",
   "devDependencies": {
-    "babel-core": "^6.21.0",
+    "@babel/core": "^7.0.0-beta.44",
+    "@babel/preset-env": "^7.0.0-beta.44",
+    "@babel/preset-es2017": "^7.0.0-beta.44",
+    "@babel/preset-react": "^7.0.0-beta.44",
+    "@babel/preset-stage-3": "^7.0.0-beta.44",
+    "babel-core": "^7.0.0-0",
```
Comment (adnasa) [inline package.json:42]: a dangling dependecy from `babel-jest` 😢

**Review by vutran — APPROVED** [review]

**Pull request merged**